### PR TITLE
[trust] Pianissimo trust fix and trust cool down adjustments

### PR DESCRIPTION
**Pull request by Omnione**
This PR is to fix two issues:
1: Enable Pianissimo to be used on trust.
fixes issue #1348 

2: Allow a party leader with a trust cool down to change leader and the new leader can cast trust right away if they currently have no cool down themselves.
fixes issue #800

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Review by Xaver-DaRed — DISMISSED** [review]
Looks good to me

File: src/map/entities/charentity.cpp [hunk 1]
```diff
@@ -234,6 +234,8 @@ CCharEntity::CCharEntity()
 
     m_Substate = CHAR_SUBSTATE::SUBSTATE_NONE;
 
+    m_TimeLastMemberJoined = server_clock::now() - std::chrono::seconds(120); // default time
```
Comment (zach2good) [inline src/map/entities/charentity.cpp]: Now that this lives in char entity, this could have a better name

**Review by zach2good — APPROVED** [review]
LGTM, consider renaming that var

File: src/map/entities/charentity.cpp (hunk 1 above)
Comment (Omnione) [reply src/map/entities/charentity.cpp]: Had a small discussion over this, the new var name seems to fit better.

File: src/map/entities/charentity.h [hunk 2]
```diff
@@ -366,6 +366,8 @@ class CCharEntity : public CBattleEntity
 
     uint32 m_LastYell;
 
+    time_point LeaderCreatedPartyTime; // Time that a party member joined and this player was leader.
```
Comment (Xaver-DaRed) [inline src/map/entities/charentity.h]: Shouldnt this be m_LeaderCreatedPartyTime?

File: src/map/party.cpp [hunk 3]
```diff
@@ -499,7 +499,7 @@ void CParty::AddMember(CBattleEntity* PEntity)
     if (PEntity->objtype == TYPE_PC && this->members.size() > 1)
     {
         auto* PLeader = dynamic_cast<CCharEntity*>(CParty::GetLeader());
-        PLeader->m_TimeLastMemberJoined = server_clock::now();
+        PLeader->LeaderCreatedPartyTime = server_clock::now();
```
Comment (Xaver-DaRed) [inline src/map/party.cpp]: m_LeaderCreatedPartyTime?

File: src/map/party.cpp [hunk 4]
```diff
@@ -1217,7 +1217,7 @@ uint32 CParty::GetTimeLastMemberJoined()
 {
     auto* PLeader = dynamic_cast<CCharEntity*>(CParty::GetLeader());
 
-    return (uint32)std::chrono::time_point_cast<std::chrono::seconds>(PLeader->m_TimeLastMemberJoined).time_since_epoch().count();
+    return (uint32)std::chrono::time_point_cast<std::chrono::seconds>(PLeader->LeaderCreatedPartyTime).time_since_epoch().count();
```
Comment (Xaver-DaRed) [inline src/map/party.cpp]: m_LeaderCreatedPartyTime?

**Review by Xaver-DaRed — CHANGES_REQUESTED** [review]
Gotta fix this

File: src/map/party.cpp (hunk 4 above)
Comment (Omnione) [reply src/map/party.cpp]: Honestly have no idea what happend here but glad you spotted it.

File: src/map/party.cpp [hunk 5]
```diff
@@ -498,7 +498,8 @@ void CParty::AddMember(CBattleEntity* PEntity)
 
     if (PEntity->objtype == TYPE_PC && this->members.size() > 1)
     {
-        this->m_TimeLastMemberJoined = server_clock::now();
+        auto* PLeader = dynamic_cast<CCharEntity*>(CParty::GetLeader());
+        PLeader->m_LeaderCreatedPartyTime = server_clock::now();
```
Comment (zach2good) [inline src/map/party.cpp]: There needs to be a check here, PLeader can be nullptr:
```cpp
auto* PLeader = dynamic_cast<CCharEntity*>(CParty::GetLeader());
if (PLeader)
{
    ... use PLeader ...
}
```

File: src/map/party.cpp [hunk 6]
```diff
@@ -1214,7 +1215,9 @@ void CParty::SetPartyNumber(uint8 number)
 
 uint32 CParty::GetTimeLastMemberJoined()
 {
-    return (uint32)std::chrono::time_point_cast<std::chrono::seconds>(m_TimeLastMemberJoined).time_since_epoch().count();
+    auto* PLeader = dynamic_cast<CCharEntity*>(CParty::GetLeader());
```
Comment (zach2good) [inline src/map/party.cpp:1222]: Same here

File: src/map/entities/charentity.cpp [hunk 7]
```diff
@@ -234,6 +234,8 @@ CCharEntity::CCharEntity()
 
     m_Substate = CHAR_SUBSTATE::SUBSTATE_NONE;
 
+    m_LeaderCreatedPartyTime = server_clock::now() - std::chrono::seconds(120); // default time
```
Comment (zach2good) [inline src/map/entities/charentity.cpp]: Why is this being set on character creation? (log in and zoning)

Comment (Omnione) [reply src/map/entities/charentity.cpp]: mistake on my part, thought we needed it, removed now

**Review by Xaver-DaRed — APPROVED** [review]

**Review by zach2good — APPROVED** [review]

**Review by claywar — APPROVED** [review]

**Pull request merged**